src/backend/distributed/executor/insert_select_executor.c

```diff
@@ -144,8 +164,102 @@ CoordinatorInsertSelectExecScanInternal(CustomScanState *node)
 			LockPartitionRelations(targetRelationId, RowExclusiveLock);
 		}
 
-		if (insertSelectQuery->onConflict || hasReturning)
+		/*
+		 * We don't support insert/select with repartitioning if:
+		 *  (*) Query is not distributed or requires merge on coordinator, or
+		 *  (*) Has returning, or
+		 *  (*) Has ON CONFLICT
```

Comment:
What's the issue with ON CONFLICT?

Reply:
No issues, `multi_insert_select_conflict` was failing because I hadn't added casts for SELECT target list, so rows weren't routed correctly and it was failing. For example, if target relation was partitioned by hash(int), we were partitioning the SELECT correctly by hash(numeric).

Fixed that and enabled ON CONFLICT.